src/px-script/main/sheet/index.js

```diff
@@ -65,98 +65,109 @@ class Sheet {
     }
 }
 
-const sheet = (() => {
-    const init = () => {
-        const sheetEls = [...document.querySelectorAll(SELECTORS.SHEET)];
+const init = ids => {
+    const sheetSelector = ids || ids === "" ? getElementsByIds(ids, "sheet") : document.querySelectorAll(SELECTORS.SHEET);
+    let sheetEls = [];
 
-        window.px._sheets = window.px._sheets || [];
+    if (sheetSelector.length) {
+        sheetEls = [...sheetSelector].map(sheet => {
+            const sheetInstance = new Sheet(sheet);
 
-        if (sheetEls.length) {
-            sheetEls.forEach(sheet => window.px._sheets.push(new Sheet(sheet)));
+            if (_sheets.length) {
+                _sheets.forEach((arraySheet, index) => {
+                    arraySheet.id === sheetInstance.id ? _sheets[index] = sheetInstance : _sheets.push(sheetInstance);
+                });
+            } else {
+                _sheets.push(sheetInstance);
+            }
 
-            // Init open buttons
-            document.querySelectorAll(SELECTORS.OPEN).forEach(btn => {
-                const id = btn.dataset.sheetOpen;
-                let sheet;
+            return sheetInstance;
+        });
 
-                window.px._sheets.forEach(s => s.id === id ? sheet = s : null);
+        // Close sheet on esc
+        document.addEventListener("keydown", e => {
+            if (e.keyCode === 27 && document.body.classList.contains("sheet-open")) {
+                handleScrollbar();
+                _sheets.forEach(sheet => sheet.isOpen ? sheet.close() : null);
+            }
+        });
 
-                if (sheet) {
-                    btn.addEventListener("click", e => {
-                        e.preventDefault();
-                        sheet.open();
-                    });
-                } else {
-                    console.warn(`OPEN: No sheet with with id ${id} was found, make sure the attribute data-sheet-open contains the correct id.`);
-                }
-            });
+        // Init open buttons
+        document.querySelectorAll(SELECTORS.OPEN).forEach(btn => {
+            const id = btn.dataset.sheetOpen;
+            let sheet;
 
-            // Init close buttons
-            document.querySelectorAll(SELECTORS.CLOSE).forEach(btn => {
-                const id = btn.dataset.sheetClose;
-                let sheet;
-
-                window.px._sheets.forEach(s => s.id === id ? sheet = s : null);
-
-                if (sheet) {
-                    btn.addEventListener("click", e => {
-                        e.preventDefault();
-                        sheet.close();
-                    });
-                } else {
-                    console.warn(`CLOSE: No sheet with with id ${id} was found, make sure the attribute data-sheet-close contains the correct id.`);
-                }
-            });
+            _sheets.forEach(s => s.id === id ? sheet = s : null);
```

Comment:
Is this a global?

Reply:
No, it is defined in the file. Moved declaration to the top instead of bottom of the file.